doc/main/tbb_userguide/Migration_Guide/Task_API.rst

```diff
@@ -325,3 +325,66 @@ task:
         }
         tg.wait();
     }
+
+Deferred task creation
+----------------------
+The low-level task API of TBB separates task creation from actual spawning. This separation allows to
+postpone task spawning while having parent task blocked(and final result production) from premature leave. 
+For example, ``RootTask``, ``ChildTask``, and ``CallBackTask`` are the user-side functors that
+inherit ``tbb::task`` and implement its interface. Then blocking of ``RootTask`` from premature leave 
+and waiting on it is implemented as: 
+
+.. code:: cpp
+
+    #include <tbb/task.h>
+
+    int main() {
+        // Assuming RootTask, ChildTask, and CallBackTask are defined.
+        RootTask& root = *new(tbb::task::allocate_root()) RootTask{};
+
+        ChildTask&    child    = *new(root.allocate_child()) ChildTask{/*params*/};
+        CallBackTask& cb_task  = *new(root.allocate_child()) CallBackTask{/*params*/};
+
+        root.set_ref_count(3);
+        
+        tbb::task::spawn(child);
+        
+        register_callback([cb_task&](){
+            tbb::task::enqueue(cb_task);
+        });
+
+        root.wait_for_all();
+        // Control flow will rich here only after both ChildTask and CallBackTask are executed,
+        // i.e. aftet the callback is called  
+    }
+
+In oneTBB this can be done using the preview feature of ``oneapi::tbb::task_group``.
+
+.. code:: cpp
+
+    #define TBB_PREVIEW_TASK_GROUP_EXTENSIONS 1
+    #include <oneapi/tbb/task_group.h>
+
+    int main(){
+        oneapi::tbb::task_group tg;
+        oneapi::tbb::task_arena arena;
+        // Assuming ChildTask, and CallBackTask are defined.
+        
+        auto cb = tg.defer(CallBackTask{/*params*/});
+        
+        register_callback([&tg, c = std::move(cb), &arena]{
+            arena.enqueue(c);
+        });        
+
+        tg.run(ChildTask{/*params*/});
+
+ 
+        tg.wait();
+        // Control flow will get here once both ChildTask CallBackTask are executed
+        // i.e. aftet the callback is called  
+    }
+
+Here ``oneapi::tbb::task_group::defer`` adds a new task into the ``tg``. Howether the task is not spawned until 
+``oneapi::tbb::task_arena::enqueue`` is called. 
+Please note that call to ``oneapi::tbb::task_group::wait`` will not return control untiol both ``ChildTask`` and 
```

Comment:
.. note::
  The call to ``oneapi::tbb::task_group::wait`` does not return the control until both ``ChildTask`` and ``CallBackTask`` 
  are executed.

Reply:
added   ``  .. note :: ``